docs/proposals/202011_vertical_block_sharding.md

```diff
@@ -0,0 +1,108 @@
+While there is work to lift this limitation and allow bigger indexes, there will be **some** other limit anyway.
+* Compaction is not streaming bytes directly from & to object storage (yet). This is due to its current index format, which is [being worked on](https://github.com/thanos-io/thanos/issues/3389).
+This means that Compactor has to download all source blocks & upload all output block files to and from disk, impacting the disk storage you need to prepare for Compactor.
+This is a **potential scalability limit for Compactor**
+* Compactor can be currently horizontally scaled to compact among streams of blocks. With some locking mechanisms we can get up to single compaction.
+However, with single block having arbitrary size, this can be **another potential scalability issue**, where you cannot split this operation to two separate machines and do concurrently.
+* Similarly, with reading the data from block, Store Gateway can shard up to single block, so with arbitrary large block, you cannot split lookup and reading to multiple
+instances with the current implementation. This poses **potential scalability issue for Store Gateway**.
+* The object storages are designed for big files, but with some limitation. The current limit e.g for [GCS is 5TB per upload/stored file](https://cloud.google.com/storage/quotas#:~:text=There%20is%20a%20maximum%20size,request%20is%20also%205%20TB).
+* Last, but not the least operating with such big blocks is not easy. We are still creating new ways of deleting, cleaning up or analysing your data for
+easier operation of Thanos. Operating (downloading, rewriting) TBs of bytes per blocks hits limitation of different tooling (e.g SRE can't just use
+own laptop anymore). 
+
+To sum up: Current logic of not caring about the block size got us far, however as expected, it's time to think about limiting the block size (primarily
+index size).
+
+### Related Tickets:
+
+* [Limit size of blocks](https://github.com/thanos-io/thanos/issues/3068)
+* [index exceeding max size of postings](https://github.com/thanos-io/thanos/issues/1424)
+
+## Goals
+
+* Automatically cap the size of index per block to X GB without impacting read performance.
+  * Optional: CLI for vertical sharding of old blocks. 
+
+## Non Goals
+
+* Redesign Block index format 
+* Introduce breaking changes
+* Lift current limitation of index size
+* Automatically shard old blocks
+
+## Proposal
+
+**We propose to vertically shard all blocks which goes beyond the specified index size limit during the compaction process.**
+
+* Special metadata in meta.json indicating a shard.
+
+TBD
+
+
+
+## Alternatives
+
+* Shard number stored in external label.
+
+## Future Work
+
+* Add [`Cuckoofilter`](https://github.com/seiflotfy/cuckoofilter) to improve query performance on store GW as well with vertically sharded blocks.
+
+![Why Cuckoo filter: https://bdupras.github.io/filter-tutorial/](../img/cuckoo-pros.png)
```

Comment:
This image is missing?  https://bdupras.github.io/filter-tutorial/ link works :+1: